File: Baekjoon/Class3/40.1931 회의실 배정.js
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").trim().split("\n");

const n = Number(input[0]);

const meetings = [];
for(let i = 1; i <= n; i++)
    meetings.push(input[i].split(" ").map(Number));
meetings.sort((a, b) => a[1] - b[1] || a[0] - b[0]); // 끝나는 시간 기준 정렬

let maxCount = 0;
let prevEnd = 0;
for(const [start, end] of meetings)
    if(prevEnd <= start) { // 다음 미팅 추가
        maxCount++;
        prevEnd = end;
    }

console.log(maxCount);

File: Baekjoon/Class3/40.1931 회의실 배정.py
import sys
input = sys.stdin.readline

n = int(input())
meetings = [tuple(map(int, input().split())) for _ in range(n)]
meetings.sort(key=lambda meeting: (meeting[1], meeting[0])) #끝나는 시간 기준 정렬

max_count = 0
prev_end = 0
for start, end in meetings:
    if prev_end <= start: #다음 미팅 추가
        max_count += 1
        prev_end = end

print(max_count)